### ISSUE TITLE
Generating Streetviewpublish1 fails (fixed), but 'cargo build' still fails.

### ISSUE THREAD

**Issue by MTRNord** [Byron/google-apis-rs#174]
Command run: ```make regen-apis```

Log:

```
Generating streetviewpublish1


Traceback (most recent call last):
  File "etc/bin/mako-render", line 324, in cmdline
    result = post_processor(template.render(**data_converted), output_file)
  File "/home/marcel/google-apis-rs/.pyenv-Linux/lib/python2.7/site-packages/mako/template.py", line 462, in render
    return runtime._render(self, self.callable_, args, data)
  File "/home/marcel/google-apis-rs/.pyenv-Linux/lib/python2.7/site-packages/mako/runtime.py", line 838, in _render
    **_kwargs_for_callable(callable_, data))
  File "/home/marcel/google-apis-rs/.pyenv-Linux/lib/python2.7/site-packages/mako/runtime.py", line 873, in _render_context
    _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
  File "/home/marcel/google-apis-rs/.pyenv-Linux/lib/python2.7/site-packages/mako/runtime.py", line 899, in _exec_template
    callable_(context, *args, **kwargs)
  File "src/mako/api/README.md.mako", line 1, in render_body
    <%
  File "/home/marcel/google-apis-rs/src/mako/lib/util.py", line 772, in new_context
    _sta_map, _fqan_map = build_activity_mappings(data_source)
  File "/home/marcel/google-apis-rs/src/mako/lib/util.py", line 673, in build_activity_mappings
    tn = to_rust_type(schemas, None, None, t, allow_optionals=False)
  File "/home/marcel/google-apis-rs/src/mako/lib/util.py", line 363, in to_rust_type
    raise AssertionError("%s: unknown dict layout: %s" % (str(err), t))
AssertionError: 'NoneType' object has no attribute 'split': unknown dict layout: {u'description': u'Request to update the metadata of photos.\nUpdating the pixels of photos is not supported.', u'type': u'object', u'properties': {u'updatePhotoRequests': {u'items': {u'properties': {u'photo': {u'description': u'Required. Photo object containing the\nnew metadata. Only the fields specified in\nupdateMask\nfield are used. If `updateMask` is not present, the update applies to all\nfields. <aside class="note"><b>Note:</b> To update\nPose.altitude,\nPose.latLngPair has to be\nfilled as well. Otherwise, the request will fail.', u'$ref': u'Photo'}, u'updateMask': {u'type': u'string', u'description': u'Mask that identifies fields on the photo metadata to update.\nIf not present, the old Photo metadata will be entirely replaced with the\nnew Photo metadata in this request. The update fails if invalid fields are\nspecified. Multiple fields can be specified in a comma-delimited list.\n\nThe following fields are valid:\n\n* `pose.heading`\n* `pose.latlngpair`\n* `pose.pitch`\n* `pose.roll`\n* `pose.level`\n* `pose.altitude`\n* `connections`\n* `places`\n\n\n<aside class="note"><b>Note:</b> Repeated fields in\nupdateMask\nmean the entire set of repeated values will be replaced with the new\ncontents. For example, if\nupdateMask\ncontains `connections` and\ngoogle.streetview.publish.v1.UpdatePhotoRequest.photo.connections is\nempty, all connections will be removed.</aside>', u'format': u'google-fieldmask'}}, u'type': u'object', u'id': u'UpdatePhotoRequest', u'description': u'Request to update the metadata of a\nPhoto. Updating the pixels of a photo\nis not supported.'}, u'type': u'array', u'description': u'Required. List of\nUpdatePhotoRequests.'}}, u'id': u'BatchUpdatePhotosRequest'}
make: *** [.api.deps:353: gen/streetviewpublish1/.timestamp] Error 1
```

**Comment by Byron** [comment]
Please note that after an update of the JSON files, it does generate correctly, even though it doesn't build just yet due to some type issues.
For that reason, it's currently on the ignore list (see `etc/api/shared.yaml`).
It should be possible to fix the generator to handle this case.

**Comment by Byron** [comment]
I am closing this issue as since then a lot has changed. All crates are now `async`, available as v2.0 on crates.io and build against the latest API spec.
Please let me know in the comments if the issue is still relevant though.
